### PR TITLE
mds: just respawn mds daemon when osd op requests timeout

### PR DESCRIPTION
**Pull request by lxbsz**
Fixes: https://tracker.ceph.com/issues/51280
Signed-off-by: Xiubo Li <xiubli@redhat.com>

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: src/mds/MDSContext.cc [hunk 1]
```diff
@@ -107,8 +107,11 @@ void MDSIOContextBase::complete(int r) {
     return;
   }
 
-  if (r == -CEPHFS_EBLOCKLISTED) {
-    derr << "MDSIOContextBase: blocklisted!  Restarting..." << dendl;
+  // It's possible that the osd op requests will be stuck and then times out
+  // after "rados_osd_op_timeout", the mds won't know what we should it, just
+  // respawn it.
+  if (r == -CEPHFS_EBLOCKLISTED || r == -CEPHFS_ETIMEDOUT) {
+    derr << "MDSIOContextBase: blocklisted or timedout!  Restarting..." << dendl;
```
Comment (batrick) [inline src/mds/MDSContext.cc:114]: For debugging purposes the difference is significant. Can you add `r` to the debug message?

**Review by batrick — CHANGES_REQUESTED** [review]

File: src/mds/MDSContext.cc (hunk 1 above)
Comment (lxbsz) [reply src/mds/MDSContext.cc]: Done.

**Comment by batrick** [comment]
jenkins test make check

**Comment by batrick** [comment]
https://tracker.ceph.com/projects/cephfs/wiki/Master#2021-June-26

**Review by batrick — APPROVED** [review]

**Pull request merged**